Cargo.toml

```diff
@@ -77,6 +77,7 @@ assert-json-diff = "2.0"
 assert_cmd = "2.0"
 assert_matches = "1.5"
 async-compat = "0.2.1"
+async-http-proxy = "1.2"
```

Comment:
Being surprised to see that this has not been required sooner, notably for `c8y_auth_proxy` to connect c8y, I added a test https://github.com/thin-edge/thin-edge.io/pull/3600/commits/4391f1e1bb99a8f8143530fc307fbecd63b62022. Indeed, not required before because `c8y_auth_proxy` 's connection is opened by `reqwest`.